ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ModifyVulnReport/EditVulnReportPage.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useEffect } from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+import {
+    PageSection,
+    Title,
+    Divider,
+    Flex,
+    FlexItem,
+    Breadcrumb,
+    BreadcrumbItem,
+    Wizard,
+    Bullseye,
+    Spinner,
+} from '@patternfly/react-core';
+
+import { vulnerabilityReportsPath } from 'routePaths';
+import useReportFormValues from 'Containers/Vulnerabilities/VulnerablityReporting/forms/useReportFormValues';
+import useSaveReport from 'Containers/Vulnerabilities/VulnerablityReporting/api/useSaveReport';
+import useFetchReport from 'Containers/Vulnerabilities/VulnerablityReporting/api/useFetchReport';
+
+import PageTitle from 'Components/PageTitle';
+import BreadcrumbItemLink from 'Components/BreadcrumbItemLink';
+import ReportParametersForm from 'Containers/Vulnerabilities/VulnerablityReporting/forms/ReportParametersForm';
+import DeliveryDestinationsForm from 'Containers/Vulnerabilities/VulnerablityReporting/forms/DeliveryDestinationsForm';
+import ReportReviewForm from 'Containers/Vulnerabilities/VulnerablityReporting/forms/ReportReviewForm';
+import NotFoundMessage from 'Components/NotFoundMessage/NotFoundMessage';
+import { getReportFormValuesFromConfiguration } from '../utils';
+import ReportFormErrorAlert from './ReportFormErrorAlert';
+import ReportFormWizardFooter from './ReportFormWizardFooter';
+
+const wizardStepNames = [
+    'Configure report parameters',
+    'Configure delivery destinations (Optional)',
+    'Review and save',
+];
+
+function EditVulnReportPage() {
+    const history = useHistory();
+    const { reportId } = useParams();
+
+    const { reportConfiguration, isLoading, error } = useFetchReport(reportId);
+    const { formValues, setFormValues, setFormFieldValue, clearFormValues } = useReportFormValues();
+    const { data, isSaving, saveError, saveReport } = useSaveReport();
+
+    // When report is created, navigate to the vuln reports page
+    // @TODO: We want to change this in the future to navigate to the read-only report page to verify the details
+    useEffect(() => {
+        if (data) {
+            clearFormValues();
+            history.push(vulnerabilityReportsPath);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [data]);
```

Comment:
That's a viable suggestion. I was thinking of the pros and cons of this approach and have a few thoughts for discussion. 

I think it definitely does give us more control over what to do in the component, which is a plus. Also, we wouldn't need to listen to the state in order to do some logic, which makes things a bit simpler. And your reason for immediately acting on the promise resolve/rejection vs. waiting for react's state update is also a plus.

I think if we ever needed to use the promise result in a deeply nested component, this might make it a bit of a hassle to pass down. We'd have to pass it using props or context. I think there might also be a case where the promise can take a while to resolve, the component could unmount, and we may end up trying to navigate or update the state on an unmounted component. That might cause memory leaks and/or unexpected behaviors. 

I also took a look at how Apollo GraphQL and React Query structure their mutation hooks, and it seems like they do it the way it is now (no promise). That's one of the reasons I did it this way.

Let me know what your thoughts are on these. (cc: @vjwilson, @pedrottimark, @bradr5, @alwayshooin)

Reply:
Discussed with Dave in PM and we agreed on an approach like this:

```
const { isLoading, error, createReport } = useCreateReport({
        onCompleted: () => {
            clearFormValues();
            history.push(vulnerabilityReportsPath);
        },
});
```

The promise approach was enticing because of immediately running the logic once a successful create/save occurred. We agreed that passing an `onCompleted` callback might be more appropriate though for the following reasons:
- Neither the function `createReport` nor the hook itself needs to know anything about what happens on a successful create. This improves modularity
- We don't need the `useEffect` anymore to make the code read more clearly
- The data flow is more explicit. We can see what happens after the mutation completes right where we defined the callback
- The `useEffect` runs whenever the dependencies change, which might not always align with when our mutation completes. So this approach is better